ecdsa/party/implementation.go

```diff
@@ -46,8 +47,8 @@ type singleSigner struct {
 
 	// used as buffer for messages received before starting signing.
 	// will be consumed once signing starts.
-	messageBuffer map[partyIdIndex][]tss.ParsedMessage
-
+	messageBuffer  map[partyIdIndex][]tss.ParsedMessage
+	PartyIdToIndex map[Digest]partyIdIndex
```

Comment:
Is it msg digest to party index?

Reply:
why is it exported?

Reply:
I'm not sure why we need this map at all. Can't we just compute the index on the fly when a message arrives to reduce the complexity of state keeping? (the calculation seems quick)

Reply:
i thought it faster that way. finding the index means we hash the party ids of all participants, then sort it.